### PR TITLE
VZ-6115 Jaeger E2E Tests

### PR DESCRIPTION
**Pull request by anajosep**
Initial set of Jaeger E2E tests to test the basic installation of Jaeger Components as part of verify install suite.

### CODE REVIEW

File: tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          containers:
+            - name: hello-helidon-container
+              image: "ghcr.io/verrazzano/example-helidon-greet-app-v1:1.0.0-1-20220513221156-7da0d32"
+              env:
+                - name: "TRACING_SERVICE"
+                  value: "hello-helidon"
+                - name: "TRACING_PORT"
+                  value: "9411"
+                - name: "TRACING_HOST"
+                  value: "jaeger-operator-jaeger-collector.verrazzano-monitoring.svc.cluster.local"
```
Comment (ayaranga) [inline tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml:34]: As we are injecting jaeger agent sidecar, shouldn't this be localhost and the port be agent port?

File: tests/e2e/jaeger/jaeger_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+		// GIVEN the Jaeger Operator component is enabled,
+		// WHEN a sample application is installed,
+		// THEN the traces are found in OpenSearch Backend
+		WhenJaegerOperatorInstalledIt("should have running pods", func() {
+			Eventually(func() (bool, error) {
+				if !isJaegerOperatorEnabled() {
+					return true, nil
+				}
+				return true, nil
```
Comment (ayaranga) [inline tests/e2e/jaeger/jaeger_test.go:182]: Code for this test and the subsequent ones is yet to be added right?

File: tests/e2e/config/scripts/install-verrazzano-kind.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ spec:
         - values:
             prometheus:
               prometheusSpec:
-                replicas: 2
+                replicas: 1
```
Comment (anders-swanson) [inline tests/e2e/config/scripts/install-verrazzano-kind.yaml:23]: Why did you change prom replicas?

File: tests/e2e/jaeger/jaeger_test.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	kubeconfigPath, err := k8sutil.GetKubeConfigLocation()
+	if err != nil {
+		AbortSuite(fmt.Sprintf("Failed to get default kubeconfig path: %s", err.Error()))
+	}
+	pkg.IsJaegerOperatorEnabled(kubeconfigPath)
+	return false
+}
+
+// 'It' Wrapper to only run spec if the Jaeger operator is supported on the current Verrazzano version
+func WhenJaegerOperatorInstalledIt(description string, f func()) {
```
Comment (anders-swanson) [inline tests/e2e/jaeger/jaeger_test.go:134]: There's a t.ItMinimumVersion function now

File: tests/e2e/config/scripts/install-verrazzano-kind.yaml (hunk 3 above)
Comment (anajosep) [reply tests/e2e/config/scripts/install-verrazzano-kind.yaml:23]: I will revert this. Some of the test failures are metrics related and I was just trying to see if the additional replicas is causing test failures.

Comment (anajosep) [reply tests/e2e/config/scripts/install-verrazzano-kind.yaml]: Change is reverted.

File: tests/e2e/jaeger/jaeger_test.go (hunk 4 above)
Comment (anajosep) [reply tests/e2e/jaeger/jaeger_test.go]: Test code is using the new function.

File: tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml (hunk 1 above)
Comment (anajosep) [reply tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml:34]: The agent exposes port `5775` 6831` and `6832` which are UDP based ports used by the Jaeger client SDK. Since the test app is not using the SDK, using these ports, makes the spans to be dropped.

**Review by ayaranga — APPROVED** [review]

File: tests/e2e/jaeger/jaeger_test.go (hunk 2 above)
Comment (anajosep) [reply tests/e2e/jaeger/jaeger_test.go]: the tests are included in the AT pipeline.
